### PR TITLE
Remove calls to TestCase::at()

### PR DESCRIPTION
**Pull request by greg0ire**
It has been deprecated as of PHPUnit 9
See https://github.com/sebastianbergmann/phpunit/issues/4297

Note that we still rely on the call order here, I am not the original
author of this test but the order looks like it is pretty important for
this test, or maybe at least for understand it.
Also note that we no longer check that save() is called after all calls
to fetch().

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Annotations/CachedReaderTest.php [hunk 1]
```diff
@@ -157,53 +157,39 @@ public function testAvoidCallingFilemtimeTooMuch() : void
...
-            ->will($this->returnValue([$route1])); // Result was cached, but there was an annotation;
-        $cache
-            ->expects($this->at(1))
-            ->method('fetch')
-            ->with($this->equalTo('[C]' . $cacheKeyMethod1))
-            ->will($this->returnValue($cacheTime));
+            ->withConsecutive(
+                // first pass => cache ok for method 1
+                // we load annotations AND filemtimes for this file
+                [$this->equalTo($cacheKeyMethod1)],
```
Comment (greg0ire) [inline tests/Doctrine/Tests/Common/Annotations/CachedReaderTest.php:166]: Does somebody know the difference between using this and using `[$cacheKeyMethod1]`?

**Comment by greg0ire** [comment]
>  the order looks like it is pretty important for this test, or maybe at least for understand it.

I noticed transient tests that have to do with `fetch` before making this PR, and I just noticed another one, so maybe we should in fact get completely rid of the order? :thinking:

**Comment by greg0ire** [comment]
Cc @bastnic in case you have ideas

**Comment by greg0ire** [comment]
> What if we use a functional test about the fetch -> save relation?

Maybe that would help. What would that look like?

**Comment by SenseException** [comment]
> Maybe that would help. What would that look like?

It could be a test that is using a real cache implementation like a custom one or maybe the `ArrayCache` and then check afterwards the expected content in the `Cache` class. I haven't checked if there are already tests like that. A custom test double implementing `Cache` could contain the checks for order, but I'm more in favour for already existing classes.

**Comment by greg0ire** [comment]
Well… I restarted the build several times tonight and could not reproduce the issue, so I guess it was more transient than I thought, meaning it's probably not related to my changes. @SenseException , I made the changes you requested, please review again.

**Pull request merged**